### ISSUE TITLE
[rke2] removing all `affinity` values for agent customization (fleet or cluster agent) results in sending regular default values instead of no values

### ISSUE THREAD

**Issue by slickwarren** [rancher/dashboard#9225]
**Setup**
- Rancher version: v2.7-head(1eb478f)
- Rancher UI Extensions:n/a
- Browser type & version: chrome

**Describe the bug**

if a user selects the `custom affinity` option and removes all default values, the cluster will instead be provisioned with default values for that agent customization affinity

**To Reproduce**

provision an rke2 cluster
* under `cluster agent customization` select the `custom affinity` option and remove all of the default entries

**Result**
cluster is provisioned with default values in cluster agent -> affinity

**Expected Result**

cluster agent should not have any affinity values

**Screenshots**

**Additional context**

this is not an issue for rke1
if the user leaves 1 entry in the custom affinity entry, it will use that one entry. This is only an issue if the list is empty.

**Comment by a-blender** [comment]
@snasovich Yes, since you can modify the cluster object via the UI. Thank you!

**Comment by gaktive** [comment]
@a-blender is a UI person to work on this? Normally we put the Working label if a UI engineer is actively working on this for a milestone, so we have to groom this first before assigning.

I'll place this in Backlog to indicate that this could be picked up but we need to determine when this would land first. From all indications, it's not a big priority but we can start with Q3.

**Comment by nwmac** [comment]
@a-blender Could you confirm that you are not working on this issue and that the UI team should pick this up? You're assigned and you've put it in the working state. Thanks.

**Comment by a-blender** [comment]
@nwmac Confirmed, I am not working on this now.
